compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -402,9 +394,9 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                     }
                 }
 
-                if !matches!(unwind, StackPopUnwind::NotAllowed) && !callee_fn_abi.can_unwind {
+                if !matches!(unwind, mir::UnwindAction::Unreachable) && !callee_fn_abi.can_unwind {
                     // The callee cannot unwind.
```

Comment:
```suggestion
                if !callee_fn_abi.can_unwind {
                    // The callee cannot unwind, so force the `Unreachable` unwind handling.
```
I know this is preexisting, but this just seems more clear.